### ISSUE TITLE
Socket Rocket Upgrade to 0.7.0 in react native version 0.73

### ISSUE THREAD

**Issue by nikhiltekwani09** [facebook/react-native#44601]
### Description

Is it possible to upgrade socket rocket version to 0.7.0 in react native 0.73 itself?

Issues with older version of socket rocket
https://github.com/facebookincubator/SocketRocket/issues/648
https://github.com/facebookincubator/SocketRocket/issues/661

### Steps to reproduce

Normally build app and try to establish mqtt connection

### React Native Version

0.73.8

### Affected Platforms

Runtime - iOS

### Output of `npx react-native info`

```text
System:
  OS: macOS 14.3.1
  CPU: (10) arm64 Apple M1 Max
  Memory: 79.28 MB / 64.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 20.11.1
    path: ~/.nvm/versions/node/v20.11.1/bin/node
  Yarn:
    version: 1.22.19
    path: /opt/homebrew/bin/yarn
  npm:
    version: 10.2.4
    path: ~/.nvm/versions/node/v20.11.1/bin/npm
  Watchman:
    version: 2024.04.15.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.15.2
    path: /usr/local/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.4
      - iOS 17.4
      - macOS 14.4
      - tvOS 17.4
      - visionOS 1.1
      - watchOS 10.4
  Android SDK: Not Found
IDEs:
  Android Studio: 2022.2 AI-222.4459.24.2221.10121639
  Xcode:
    version: 15.3/15E204a
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.11
    path: /usr/bin/javac
  Ruby:
    version: 2.6.10
    path: /usr/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react: Not Found
  react-native:
    installed: 0.73.8
    wanted: 0.73.8
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: false
```

### Stacktrace or Logs

```text
${PROJECT_PATH}/ios/Pods/SocketRocket/SocketRocket/Internal/RunLoop/SRRunLoopThread.m:79 Thread running at User-interactive quality-of-service class waiting on a lower QoS thread running at Default quality-of-service class. Investigate ways to avoid priority inversions
#0	0x00000001007c1d7f in -[SRRunLoopThread runLoop] at ${PROJECT_PATH}/ios/Pods/SocketRocket/SocketRocket/Internal/RunLoop/SRRunLoopThread.m:79
#1	0x00000001007bc297 in +[NSRunLoop(SRWebSocket) SR_networkRunLoop] at ${PROJECT_PATH}/ios/Pods/SocketRocket/SocketRocket/NSRunLoop+SRWebSocket.m:24
#2	0x00000001007c06a7 in -[SRProxyConnect _openConnection] at ${PROJECT_PATH}/ios/Pods/SocketRocket/SocketRocket/Internal/Proxy/SRProxyConnect.m:287
#3	0x00000001007bfb5f in -[SRProxyConnect _configureProxy] at ${PROJECT_PATH}/ios/Pods/SocketRocket/SocketRocket/Internal/Proxy/SRProxyConnect.m:182
#4	0x00000001007bf353 in -[SRProxyConnect openNetworkStreamWithCompletion:] at ${PROJECT_PATH}/ios/Pods/SocketRocket/SocketRocket/Internal/Proxy/SRProxyConnect.m:85
#5	0x00000001007c3b3b in -[SRWebSocket open] at ${PROJECT_PATH}/ios/Pods/SocketRocket/SocketRocket/SRWebSocket.m:333
#6	0x00000001001ec343 in -[RCTReconnectingWebSocket start] at ${PROJECT_PATH}/node_modules/react-native/Libraries/WebSocket/RCTReconnectingWebSocket.m:52
#7	0x00000001001d7363 in -[RCTPackagerConnection init] at ${PROJECT_PATH}/node_modules/react-native/React/DevSupport/RCTPackagerConnection.mm:71
#8	0x00000001001d71a3 in __49+[RCTPackagerConnection sharedPackagerConnection]_block_invoke at ${PROJECT_PATH}/node_modules/react-native/React/DevSupport/RCTPackagerConnection.mm:58
#9	0x00000000000067bb in _dispatch_client_callout ()
#10	0x000000000000834b in _dispatch_once_callout ()
#11	0x00000001001d7157 in +[RCTPackagerConnection sharedPackagerConnection] at /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.4.sdk/usr/include/dispatch/once.h:85
#12	0x00000001002651d7 in -[RCTDevSettings initialize] at ${PROJECT_PATH}/node_modules/react-native/React/CoreModules/RCTDevSettings.mm:182
#13	0x00000001001c2cab in -[RCTModuleData _initializeModule] at ${PROJECT_PATH}/node_modules/react-native/React/Base/RCTModuleData.mm:272
#14	0x00000001001c2603 in -[RCTModuleData setUpInstanceAndBridge:] at ${PROJECT_PATH}/node_modules/react-native/React/Base/RCTModuleData.mm:223
#15	0x00000001001c40ff in -[RCTModuleData instance] at ${PROJECT_PATH}/node_modules/react-native/React/Base/RCTModuleData.mm:395
#16	0x00000001001699ef in -[RCTCxxBridge moduleForName:lazilyLoadIfNecessary:] at ${PROJECT_PATH}/node_modules/react-native/React/CxxBridge/RCTCxxBridge.mm:587
#17	0x00000001001ccdcf in -[RCTModuleRegistry moduleForName:lazilyLoadIfNecessary:] at ${PROJECT_PATH}/node_modules/react-native/React/Base/RCTModuleRegistry.m:39
#18	0x00000001001ccd37 in -[RCTModuleRegistry moduleForName:] at ${PROJECT_PATH}/node_modules/react-native/React/Base/RCTModuleRegistry.m:30
#19	0x0000000100275393 in -[RCTPerfMonitor devMenuItem] at ${PROJECT_PATH}/node_modules/react-native/React/CoreModules/RCTPerfMonitor.mm:151
#20	0x000000010027525f in -[RCTPerfMonitor initialize] at ${PROJECT_PATH}/node_modules/react-native/React/CoreModules/RCTPerfMonitor.mm:137
#21	0x00000001001c2cab in -[RCTModuleData _initializeModule] at ${PROJECT_PATH}/node_modules/react-native/React/Base/RCTModuleData.mm:272
#22	0x00000001001c2603 in -[RCTModuleData setUpInstanceAndBridge:] at ${PROJECT_PATH}/node_modules/react-native/React/Base/RCTModuleData.mm:223
#23	0x00000001001c4337 in __25-[RCTModuleData instance]_block_invoke at ${PROJECT_PATH}/node_modules/react-native/React/Base/RCTModuleData.mm:391
#24	0x0000000100225b03 in RCTUnsafeExecuteOnMainQueueSync at ${PROJECT_PATH}/node_modules/react-native/React/Base/RCTUtils.m:275
#25	0x00000001001c3f9f in -[RCTModuleData instance] at ${PROJECT_PATH}/node_modules/react-native/React/Base/RCTModuleData.mm:390
#26	0x000000010016e2cf in __49-[RCTCxxBridge _prepareModulesWithDispatchGroup:]_block_invoke at ${PROJECT_PATH}/node_modules/react-native/React/CxxBridge/RCTCxxBridge.mm:1002
#27	0x0000000000004b97 in _dispatch_call_block_and_release ()
#36	0x000000018a658517 in UIApplicationMain ()
#37	0x00000001000084ab in main at ${PROJECT_PATH}/ios/MessengerExample/main.m:7
#38	0x00000001ab706d83 in start ()
```

### Reproducer

https://github.com/react-native-community/reproducer-react-native

**Comment by nikhiltekwani09** [comment]
> ⚠️	Missing Reproducible Example
> ℹ️	We could not detect a reproducible example in your issue report. Please provide either:
> * If your bug is UI related: a [Snack](https://snack.expo.dev)
> * If your bug is build/update related: use our [Reproducer Template](https://github.com/react-native-community/reproducer-react-native/generate). A reproducer needs to be in a GitHub repository under your username.

This is already fixed by you guys in react native 0.74, I m just asking for support in react native 0.73

**Comment by HoseinKhanBeigi** [comment]
To upgrade SocketRocket to version 0.7.0 in React Native 0.73.x, you'll need to specify the version explicitly in your Podfile
Open your Podfile and specify the desired version of SocketRocket. Add the following line to ensure SocketRocket is included and specify the version 0.7.0:
Install CocoaPods Dependencies
Clean and Rebuild the Project

**Comment by cortinico** [comment]
Also we have SocketRocket bumped to 0.7.0 in 0.74.x - See bab9c839eaad1ffc358fa5c87440a6de7cbe981b

You should be updating your React Native version instead.
